npm/iptm/iptm.go

```diff
@@ -127,13 +128,23 @@ func (iptMgr *IptablesManager) UninitNpmChains() error {
 		return err
 	}
 
-	// For backward compatibility, we should be cleaning older chains.
-	// TODO(jungukcho): need to check K8s or NPM version and do it selectively
-	// to avoid unnecessary call.
+	// Clean old NPM chains. This is forward compatible with NPM v2.
 	allAzureChains := append(IptablesAzureChainList,
 		util.IptablesAzureTargetSetsChain,
 		util.IptablesAzureIngressWrongDropsChain,
 	)
+	currentAzureChains, err := linuxutil.AllCurrentAzureChains(iptMgr.exec, defaultlockWaitTimeInSeconds)
+	if err != nil {
+		metrics.SendErrorLogAndMetric(util.IptmID, "Error: failed to get all current AZURE-NPM chains")
+	} else {
+		// add any extra current azure chains to the list of all azure chains.
+		for _, chain := range allAzureChains {
+			delete(currentAzureChains, chain)
+		}
+		for chain := range currentAzureChains {
+			allAzureChains = append(allAzureChains, chain)
```

Comment:
We are adding indices here and not the element at that index. Is that what you intend to do?